### PR TITLE
Add setting to enable single-click navigation in file browser

### PR DESCRIPTION
**Pull request by RRosio**
## References

Adding a setting in file-browser extension to allow single-click navigation as per https://github.com/jupyterlab/jupyterlab/issues/14071.

## Code changes

- [x] add command to toggle single-click navigation
- [x] update logic to allow single-click navigation

## User-facing changes

Users can access the setting option by going to the Settings Editor -> File Browser and select the option `Navigate files and directories with single click` 

## Backwards-incompatible changes
N/A

### CODE REVIEW

File: packages/filebrowser-extension/schema/browser.json [hunk 1]
```diff
@@ -28,6 +28,15 @@
...
         ]
+      },
+      {
+        "id": "jp-mainmenu-view",
+        "items": [
+          {
+            "command": "filebrowser:toggle-single-click",
+            "rank": 9.99
+          }
+        ]
```
Comment (krassowski) [inline packages/filebrowser-extension/schema/browser.json:39]: Do we really want this in the main menu? I would say a setting would be enough.

**Comment by krassowski** [comment]
I pushed a commit after running `jlpm lint` locally. Maybe you need to re-install the pre-commit hooks? What does `jlpm --version` return?

**Comment by krassowski** [comment]
User-facing documentation would be a nice enhancement but would not block this PR

File: packages/filebrowser/src/listing.ts [hunk 2]
```diff
@@ -1025,6 +1025,18 @@ export class DirListing extends Widget {
...
+   * This enables opening files/directories with a single click.
+   */
+  setAllowSingleClickNavigation(isEnabled: boolean) {
+    this._allowSingleClick = isEnabled;
+  }
+
+  getAllowSingleClickNavigation() {
+    return this._allowSingleClick;
+  }
+
```
Comment (krassowski) [inline packages/filebrowser/src/listing.ts:1039]: I think we do not need it; let's keep the public API to minimum here.

```suggestion
```

File: packages/filebrowser/src/listing.ts [hunk 3]
```diff
@@ -1138,6 +1150,10 @@ export class DirListing extends Widget {
       document.addEventListener('mouseup', this, true);
       document.addEventListener('mousemove', this, true);
     }
+
+    if (this.getAllowSingleClickNavigation()) {
```
Comment (krassowski) [inline packages/filebrowser/src/listing.ts:1154]: Allows to reduce the API surface, and removes one function call :)

```suggestion
    if (this._allowSingleClick) {
```

**Review by krassowski — COMMENTED** [review]
Some minor suggestions on namig

**Review by krassowski — APPROVED** [review]
Thank you @RRosio! Tested locally, works great. Selecting multiple files without opening them is still possible with <kbd>Ctrl</kbd> + :computer_mouse: click.

**Comment by krassowski** [comment]
A great follow up here would be adding a test because otherwise this feature may break with any future refactoring.

**Comment by jtpio** [comment]
> A great follow up here would be adding a test because otherwise this feature may break with any future refactoring.

Opened https://github.com/jupyterlab/jupyterlab/issues/16640 to track this.

**Review by jtpio — APPROVED** [review]
Thanks @RRosio @krassowski !

**Pull request merged**